### PR TITLE
WIP #16

### PR DESCRIPTION
**Pull request by angelicambg**

### CODE REVIEW

File: R/test-data.R [hunk 1]
```diff
@@ -228,19 +278,40 @@ get_daily_test_data <- function() {
...
-    dplyr::arrange(date, country) %>%
-    dplyr::relocate(country, tests_cumulative, new_tests, date, source) %>%
-    dplyr::select(
-      country, tests_cumulative, new_tests,
-      tests_cumulative_corrected, new_tests_corrected, date, source
+    dplyr::arrange(date, country)
+
+  test_combined_split <- test_combined %>%
+    dplyr::group_split(date)
+  # The firs date didn't change
```
Comment (pat-s) [inline R/test-data.R]: Can we be more verbose on what is happening here?

File: R/test-data.R [hunk 2]
```diff
@@ -228,19 +278,40 @@ get_daily_test_data <- function() {
...
       tests_cumulative_corrected
     )) %>%
-    dplyr::arrange(date, country) %>%
-    dplyr::relocate(country, tests_cumulative, new_tests, date, source) %>%
-    dplyr::select(
-      country, tests_cumulative, new_tests,
-      tests_cumulative_corrected, new_tests_corrected, date, source
+    dplyr::arrange(date, country)
+
+  test_combined_split <- test_combined %>%
```
Comment (pat-s) [inline R/test-data.R]: Maybe we can briefly document why this is done.

File: R/test-data.R [hunk 3]
```diff
@@ -154,72 +154,122 @@ process_test_data <- function() {
...
+        dplyr::relocate(
+          country, tests_cumulative, new_tests,
+          tests_cumulative_corrected, new_tests_corrected,
+          date, source
+        )
     },
     error = function(cond) {
       cli::cli_alert_info("No file with manual test countries found for
-          today. Ignoring input.", wrap = TRUE)
+                today. Ignoring input.", wrap = TRUE)
```
Comment (pat-s) [inline R/test-data.R]: I think this message (from me) could be improved with respect to wording :)

File: R/test-data.R [hunk 4]
```diff
@@ -154,72 +154,122 @@ process_test_data <- function() {
...
+        owner = "dsbbfinddx",
+        repo = "FINDCov19TrackerData",
+        branch = "selenium"
+      )
+
+      filelist_manual <- unlist(lapply(fl_gh$tree, "[", "path"), use.names = FALSE) %>%
+        stringr::str_subset(., "manual/processed/.*processed-manually.csv$") %>%
+        paste0("https://raw.githubusercontent.com/dsbbfinddx/FINDCov19TrackerData/master/", .)
+
+      manual_files <- manual_list[which(manual_list %in% filelist_manual)]
```
Comment (pat-s) [inline R/test-data.R]: I think this part could also benefit from some brief comments.

File: tests/testthat/test-test-data.R [hunk 5]
```diff
@@ -1,10 +1,12 @@
-test_that("get_daily_test_data() works as expected", {
-  get_daily_test_data()
+test_that("get_test_data() works as expected", {
+  get_test_data()
```
Comment (pat-s) [inline tests/testthat/test-test-data.R]: I think we could use `write = FALSE` here and just perform checks on the returned object.

**Review by pat-s — COMMENTED** [review]
Thanks!
Looks promising on the first look.
Do we need to add some tests which adapt to the new behavior?

I think adding some examples in the roxygen function documentation with different usages of the function (i.e. `write = FALSE`) could be benefitial.

Also the overall description of the function probably needs an update now.

**Comment by angelicambg** [comment]
@pat-s I did changes based on the comments. Could you take a look?

File: R/helper.R [hunk 6]
```diff
@@ -36,65 +36,65 @@ calculate_daily_tests_r_fetch <- function(data, tests_cumulative) {
...
+#   data_comb <- dplyr::left_join(data, data_yesterday, by = "country")
+#
+#   data_new_tests <- data_comb %>%
+#     dplyr::mutate(new_tests = tests_cumulative.x - tests_cumulative.y) %>%
+#     dplyr::rename(tests_cumulative = tests_cumulative.x, date = date.x) %>%
+#     dplyr::relocate(country, tests_cumulative, new_tests, date) %>%
+#     dplyr::select(-tests_cumulative.y, -date.y)
+#
+#   return(data_new_tests)
+# }
```
Comment (pat-s) [inline R/helper.R]: Just remove old code - no need to keep commented outdated code :)

File: R/test-data.R [hunk 7]
```diff
@@ -154,72 +156,139 @@ process_test_data <- function() {
...
+          country, tests_cumulative, new_tests,
+          tests_cumulative_corrected, new_tests_corrected,
+          date, source
+        )
     },
     error = function(cond) {
-      cli::cli_alert_info("No file with manual test countries found for
-          today. Ignoring input.", wrap = TRUE)
+      cli::cli_alert_info("Any file founded in folder manual/processed for the
+              input days. Ignoring uploaded manual tests.", wrap = TRUE)
```
Comment (pat-s) [inline R/test-data.R]: How about "no file found [...] for days $days. Ignoring manual test files for these days."?

**Review by pat-s — COMMENTED** [review]
Thanks! 

FYI: You can mark pull requests as "draft state". With this you do not need to edit the PR title and can keep a descriptive title.

**Pull request merged**